channeldb/migrations.go

```diff
@@ -573,3 +573,40 @@ func migratePruneEdgeUpdateIndex(tx *bolt.Tx) error {
 
 	return nil
 }
+
+// migrateOptionalChannelCloseSummaryFields migrates the serialized format of
+// ChannelCloseSummary to a format where optional fields' presence is indicated
+// with boolean markers.
+func migrateOptionalChannelCloseSummaryFields(tx *bolt.Tx) error {
```

Comment:
would be nice to have a unit test for this migration, asserting it behaves properly in all four cases

Reply:
Migration added! (only three cases though?)